### PR TITLE
use new non-extensions-based panel, remove old HierarchicalLocationsView

### PR DESCRIPTION
**Pull request by sqs**
The new panel was introduced by the coolCodeIntel experimental feature flag and has defaulted to `on` since 2022-09-21 (commit e6b6bc0ec82534378a57bda5033bfaa970a3890d). It replaces the old multi-column UI with separate definitions and references panels.

## Test plan

Run "Find references" and ensure the new panel still works.

### CODE REVIEW

**Comment by sqs** [comment]
cc @mrnugget @olafurpg

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| -2.47% (-68.35 kb) 🔽 | -0.27% (-38.98 kb) 🔽 | 0.25% (+29.38 kb) 🔺 | 1.66% (+12) 🔺 |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/sqs/rm-extension-api-etc/rm-old-panel/compare-report.html;tab=live_object) for a full comparison between the commits 4664a17 and 323638f or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: schema/settings.schema.json [hunk 1]
```diff
@@ -467,11 +467,6 @@
         }
       }
     },
-    "codeIntel.referencesPanel": {
-      "description": "What kind of references panel to use for exploring definitions and references. The tabbed panel is deprecated.",
-      "type": "string",
-      "enum": ["tree-view", "tabbed"]
-    },
```
Comment (mrnugget) [inline schema/settings.schema.json:474]: If I understand it correctly this was added specifically for some folks who complained and wanted the old panel back.

I totally get if we say "only 1 panel now" (and that's what I personally would've done), but just want to call that out.

Comment (sqs) [reply schema/settings.schema.json:474]: Yeah, that is the intent ("only 1 panel now"), but I'll go through and find out why people complained.

Comment (olafurpg) [reply schema/settings.schema.json:474]: We received several complains from different customers about UX regressions in the new ref panel. cc/ @jasonhawkharris who was in direct contact with some of these customers. 

If we want to play the "only 1 panel now" card then I think we first need to listen more carefully to valid complaints about the new panel.

Some of these regressions have been fixed but not all of them. At the top of my head, the most important regression is that it's more complicated to promote a reference item into the main blob with the new panel. Most people think they have to click on the item first, move the mouse into the preview and then click on a line in the preview (compared to clicking once on a reference item in the old panel).

CleanShot 2023-01-04 at 10 21 17@2x

I think we can fix that regression by giving a stronger UI hint that you can click twice on the tree item itself without moving the mouse into the preview. I've heard from several people internally that they didn't know this is possible.

Comment (olafurpg) [reply schema/settings.schema.json:474]: My understanding is that we should never remove experimental settings in the schema, we should only update the docstring to indicate which values are no longer supported and therefore have no effect.

Comment (limitedmage) [reply schema/settings.schema.json:474]: > My understanding is that we should never remove experimental settings in the schema, we should only update the docstring to indicate which values are no longer supported and therefore have no effect.

@olafurpg this was true until #44180 was fixed :D Now we can safely remove items from the settings schema.

Comment (philipp-spiess) [reply schema/settings.schema.json:474]: > I think we can fix that regression by giving a stronger UI hint that you can click twice on the tree item itself without moving the mouse into the preview. I've heard from several people internally that they didn't know this is possible.

Oh wow, I also had no idea that this existed. I guess we can try to add an icon to the left of the line numbers or so? I’m thinking of something like the icons you often see for "external" links so it's clear that something _outside_ the ref panel is happening. I'll look into it.

**Review by philipp-spiess — APPROVED** [review]
Amazing cleanup! Stamping for the code review and I agree with the others here that it makes sense to remove this feature. We should prfioritize fixing the issues that customers report to make the experience of the new code intel panel better for everyone. I'll look into the issue that @olafurpg pointed out above.

**Comment by sqs** [comment]
Based on @philipp-spiess 's comment (https://github.com/sourcegraph/sourcegraph/pull/46052#pullrequestreview-1240706154), I'm going to merge this. Thanks!

**Pull request merged**